src/PluginSuppression.php

```diff
@@ -87,13 +96,22 @@ function ( $props, $block_name ) {
 			2
 		);
 
+		// Priority 8 needed to run before ReaderThemeLoader::override_theme() at priority 9.
+		add_action( 'plugins_loaded', [ $this, 'initialize' ], 8 );
+	}
+
+	/**
+	 * Initialize.
+	 */
+	public function initialize() {
 		// When a Reader theme is selected and an AMP request is being made, start suppressing as early as possible.
 		// This can be done because we know it is an AMP page due to the query parameter, but it also _has_ to be done
 		// specifically for the case of accessing the AMP Customizer (in which customize.php is requested with the query
 		// parameter) in order to prevent the registration of Customizer controls from suppressed plugins. Suppression
 		// could be done early for Transitional mode as well since a query parameter is also used for frontend requests
 		// but there is no similar need to suppress the registration of Customizer controls in Transitional mode since
 		// there is no separate Customizer for AMP in Transitional mode (or legacy Reader mode).
+		// @todo This could rather check if ! amp_is_canonical().
```

Comment:
You mean to replace the entire condition with `! amp_is_canonical()`?

Reply:
Yes. Or rather, it could be:

```suggestion
		// @todo This check could be replaced with ( ! amp_is_canonical() && $this->paired_routing->has_endpoint() ).
```

Reply:
Yea that could work.